### DISCUSSION TITLE
cjpegli

### DISCUSSION THREAD

**Discussion by Qhilm** [Homebrew/discussions#5074] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c612baa096c69f2fcb221c74e6f5b9979efdcee
Last commit: 7 days ago
Core tap JSON: 21 Jan 12:18 UTC
Core cask tap JSON: 21 Jan 12:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

Hello,

I would like to install [cjpegli](https://github.com/libjxl/libjxl/blob/main/lib/jpegli/README.md) but I cannot find a formula for this.

I am not familiar at all with how formulae are created in homebrew, hence I'm trying to understand where I should be raising a feature enhancement request or a bug? Is it something that needs to be fixed in the libjxl project or does rather an entry have to be added on the homebrew side?

**Comment by SMillerDev** [comment]
https://docs.brew.sh/Adding-Software-to-Homebrew

**Comment by Qhilm** [comment]
Unless I misunderstood something, these instructions seem a bit outdated. I am already failing at the second command:

```
$ brew tap homebrew/core
Error: Tapping homebrew/core is no longer typically necessary.
Add --force if you are sure you need it done.
```

Obviously, I can --force, but it is a little confusing that the instructions do not mention this.

What I would also like to understand is the amount of effort that goes into maintaining this. If someone with some experience could comment on this, that would be welcome.

**Reply by SMillerDev** [reply]
> Obviously, I can --force, but it is a little confusing that the instructions do not mention this.

Could you make an issue for that in https://github.com/homebrew/brew or even better, edit https://github.com/Homebrew/brew/blob/master/docs/Adding-Software-to-Homebrew.md

**Reply by Qhilm** [reply]
PR [16542](https://github.com/Homebrew/brew/pull/16542)

**Comment by beaudh** [comment]
According to the vendors doco:
brew install jpeg-xl
https://github.com/libjxl/libjxl/blob/main/README.md

**Comment by Qhilm** [comment]
I need to figure out what the issue with the `jpeg-xl` formula is.

It seems I need to set install parameter `DJPEGXL_ENABLE_JPEGLI` to ON, but after editing the formula locally, building from source fails.

https://github.com/Homebrew/homebrew-core/blob/6b90bd87b55c14dd8cc04829be39f41e59b90b78/Formula/j/jpeg-xl.rb

This is the error I get when building from source:

```
-- Could NOT find Python3 (missing: Interpreter) 
    Reason given by package: 
        Interpreter: Cannot run the interpreter "/opt/homebrew/opt/asciidoc/libexec/bin/python3"
```

I checked, this path does not exists, even though the formula asciidoc is correctly installed.

I don't know any further I think.

**Reply by cho-m** [reply]
That message is because `asciidoc` was changed from virtualenv to `pip install` so the interpreter path is incorrect. This may change back depending on how we end up handling `asciidoc` in relation to Homebrew's new rules on Python libraries, i.e. it will probably fail the upcoming audit in https://github.com/Homebrew/brew/pull/16663

**Comment by VonZubinski** [comment]
@Qhilm I experienced the same problem and opened an issue here:
https://github.com/libjxl/libjxl/issues/3360
